src/button-group/button-group.tsx

```diff
@@ -131,6 +131,32 @@ export default class ButtonGroup extends React.Component<ButtonGroupProps> {
                         return {};
                       }
 
+                      // Set Border Radious For Buttons According To The Button Position
+                      if (index === 0) {
+                        return {
+                          borderTopRightRadius: '0px',
+                          borderBottomRightRadius: '0px',
+                        };
+                      }
+
+                      if (index !== children.length - 1) {
+                        return {
+                          borderTopRightRadius: '0px',
+                          borderBottomRightRadius: '0px',
+                          borderTopLeftRadius: '0px',
+                          borderBottomLeftRadius: '0px',
+                          marginLeft: '1px',
+                        };
+                      }
+
+                      if (index === children.length - 1) {
+                        return {
+                          borderTopLeftRadius: '0px',
+                          borderBottomLeftRadius: '0px',
+                          marginLeft: '1px',
```

Comment:
in case of shape !== default a scale100 margin is expected (line 160).

Capture the border radius values in an object in all the three cases, and append object to the below return statements. Leave margins as is.

Reply:
Below return means the return statement in line 167, right?

Reply:
both 161 and 167.